### PR TITLE
add e2e tests

### LINKED ISSUE #313: Playwright E2E Testing
As next-yak continues to grow in complexity and feature set, we need to ensure that the CSS generation and application works correctly across various contexts. (e.g: dev, prod app-router, pages-router and soon turbopack)

E2E tests will allow us to:

1. Verify compiled CSS is correctly applied to DOM elements
2. Test cross-file mixins and selectors
3. Catch regressions that might occur with Next.js version updates

## Testing Strategy

The idea would be to:

1. Start a Next.js dev or production server with test components
2. Navigate to specific test pages
3. Verify that elements have the expected computed styles
4. Test interactive behavior where applicable

### Core Functionality

1. **Basic Styling**
   - Test static styles with the `styled` API
   - Verify correct CSS class generation and application
   - Check specificity and inheritance patterns

2. **Dynamic Styles**
   - Test conditional class toggling based on props
   - Verify correct style application when props change
   - Test style override behavior

3. **Dynamic Properties**
   - Test CSS variable generation for dynamic props
   - Verify correct application of custom properties
   - Test updates to dynamic values

4. **Keyframes Animations**
   - Test keyframe animation definitions
   - Verify correct animation application
   - Test animation name generation

### Advanced Patterns

5. **Cross-File Mixins**
   - Test importing and using mixins from different files
   - Verify correct CSS generation and application
   - Test nested mixins across files

6. **Cross-File Selectors**
   - Test targeting components defined in other files
   - Verify selector specificity and cascade
   - Test complex cross-file selector patterns

7. **CSS Prop**
   - Test inline css prop usage
   - Verify composition with className and style props
   - Test dynamic values in css props

8. **Tailwind/Atomic CSS Integration**
   - Test `atoms` API with Tailwind classes
   - Verify correct class application
   - Test conditional Tailwind class application

9. **Nested Selectors**
   - Test parent-child relationships in styled components
   - Verify correct nesting of selectors
   - Test pseudo-class and pseudo-element behavior

10. **Component Targeting**
    - Test targeting styled components within other components
    - Verify hover, focus, and other interactive states
    - Test complex component selector patterns

### Edge Cases

11. **Media Queries**
    - Test responsive styles with media queries
    - Verify correct application at different viewport sizes
    - Test media query prioritization and cascade

12. **Build-Time Constants**
    - Test .yak.ts constant resolution
    - Verify correct constant usage in styles
    - Test complex constant value patterns

13. **Class Minification**
    - Test production builds with minified class names
    - Compare development and production class patterns
    - Verify correct style application with minified classes

14. **Theming Support**
    - Test theme context and application
    - Verify theme value propagation to styled components
    - Test theme switching behavior

### PR DESCRIPTION
**Pull request by jantimon**
add bundler independent testing which verifies that the browser renders the correct styles

  - styled-basic — Basic styled component rendering with CSS properties (color, font-size)                                                                                                                                                                                                                                                    
  - styled-composition — Styled component inheritance where extended components override base styles                                                                                                                                                                                                                                          
  - css-prop — The css prop with basic, conditional, and nested styles                                                                                                                                                                                                                                                                        
  - css-mixin — CSS mixins used in both styled components and css prop                                                                                                                                                                                                                                                                        
  - dynamic-props — Dynamic props with CSS variables and conditional styles                                                                                                                                                                                                                                                                   
  - attrs — .attrs() method for default attributes, dynamic attributes, and prop overrides                                                                                                                                                                                                                                                    
  - pseudo-elements — ::before and ::after pseudo-elements with content generation                                                                                                                                                                                                                                                            
  - nested-selectors — Nested CSS selectors including child combinator and pseudo-classes                                                                                                                                                                                                                                                     
  - has-selector — CSS :has() pseudo-class for parent reactivity to child state                                                                                                                                                                                                                                                               
  - media-query — Media queries both inline and imported from cross-file constants
  - keyframes-animation — CSS @keyframes animations
  - layer — CSS @layer specificity rules (unlayered styles overriding layered styles)
  - component-selector — Component selector targeting imported styled components
  - cross-file-import — Cross-file constant imports + HMR updates when dependency changes
  - cross-file-mixin — Styled component using CSS mixin imported from another file
  - namespace-import — Styled components using namespace-imported constants (import * as C)
  - nested-mixin — Nested cross-file mixins with component selectors
  - yak-file — .yak.ts file constants + HMR when they change
  - yak-file-mixin — Dynamically generated mixins from .yak.ts files with calculations
  - yak-file-with-import — HMR when transitive .ts dependency of .yak.ts file changes
  - hmr-css-change — HMR when CSS in source file changes without full page reload
  - hmr-new-component — HMR when adding a new styled component without full page reload

fixes #313

### CODE REVIEW

File: e2e/package.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "next-yak-e2e",
+  "private": true,
+  "type": "module",
+  "exports": {
+    ".": "./test-env.ts"
+  },
+  "scripts": {
+    "test": "node --experimental-strip-types run.ts",
```
Comment (Mad-Kat) [inline e2e/package.json]: this is only necessary in node <22

File: e2e/cases/media-query/queries.yak.ts
Comment (Mad-Kat) [inline e2e/cases/media-query/queries.yak.ts]: this isn't a very good example for the need of a yak file. Should we add some mapping to it?

File: e2e/package.json (hunk 1 above)
Comment (jantimon) [reply e2e/package.json]: good point 👍
